### DISCUSSION TITLE
Merri-Bek City Council (Australia) source changes

### DISCUSSION THREAD

**Discussion by superdeadguy** [mampfes/hacs_waste_collection_schedule#1930] [General]
Hi all, I found that Merri-Bek City Council in Australia must have made a change to their URL around the end of March 2024 that provides the reply to the sources query. The schedule appears in HASS as unknown. Hopefully this small fix helps someone else out.

Inspecting the logs in HASS and the sources PY file, the URL was https://www.merri-bek.vic.gov.au/gis/AddressDetails.svc/fourbins/?wasteDay=Monday&wasteRateCode=101&recycleRateCode=142&fogoRateCode=160&glassRateCode=170&zone=A&glassWeekNumber=4&address=ADDRESS&cpage=94106
The source returned "unknown" in HASS and when querying the Merri-Bek website, the final parameter "cpage=94106" is different than what the Merri-Bek website expects now. The new value is "cpage=2823". Omitting this parameter or using any other number causes a bad request response.

I edited the merri_bek_vic_gov_au.py file, in the PARAMS section to the following and it's working again after a reload.

/homeassistant/custom_components/waste_collection_schedule/waste_collection_schedule/source/merri_bek_vic_gov_au.py
        PARAMS = {
            "wasteDay": attributes["Day"],
            "wasteRateCode": attributes["Waste_Rate_Code"],
            "recycleRateCode": attributes["Recycling_Rate_Code"],
            "fogoRateCode": attributes["FOGO_Rate_Code"],
            "glassRateCode": attributes["Glass_Rate_Code"],
            "zone": attributes["Zone"],
            "glassWeekNumber": attributes["GlassWeek"],
            "address": attributes["EZI_Address"],
            "cpage": "2823",

**Comment by 5ila5** [comment]
Thanks. Implemented this fix in #1936

This will be part of release 1.47.0 or is available now as master (git) version (3-dot menu -> redownload -> change version to master -> fully restart HA (this will increase update cycle by a lot)
